site/src/components/SettingsHeader/SettingsHeader.tsx

```diff
@@ -1,63 +1,54 @@
-import { useTheme } from "@emotion/react";
 import { Button } from "components/Button/Button";
-import { Stack } from "components/Stack/Stack";
 import { SquareArrowOutUpRightIcon } from "lucide-react";
 import type { FC, ReactNode } from "react";
+import { twMerge } from "tailwind-merge";
 
-interface HeaderProps {
+type HeaderHierarchy = "primary" | "secondary";
+type HeaderLevel = `h${1 | 2 | 3 | 4 | 5 | 6}`;
+
+type HeaderProps = Readonly<{
 	title: ReactNode;
 	description?: ReactNode;
-	secondary?: boolean;
+	titleVisualHierarchy?: HeaderHierarchy;
+	titleHeaderLevel?: HeaderLevel;
```

Comment:
I know it is how we did before but since we are touching this, could we also improve the abstraction? I see something like:

```tsx
<SettingsHeader actions={[<SeeDocsButton />]}>
  <SettingsHeaderTitle>Mytitle</SettingsHeaderTitle>
  <SettingsHeaderSubtitle>Some text hear</SettingsHeaderSubtitle>
</SettingsHeader>
```